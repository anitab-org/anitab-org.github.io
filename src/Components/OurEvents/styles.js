import { View, Text, StyleSheet } from 'react-native';
import { styled } from 'react-native-reflect';

export const Box = styled(View, {
  flex: 1,
  width: '80%',
  flexDirection: 'column',
  paddingLeft: 16,
  paddingRight: 16,
  flexWrap: 'wrap',
  marginTop: 32,
});

export const Content = styled(View, {
  flexDirection: 'column',
  paddingLeft: 16,
  marginRight:77,
});

export const Description = styled(Text, {
  flex: 1,
  marginTop: 32,
  paddingLeft: 16,
  fontSize: 18,
  flexGrow: 1,
  fontWeight: 200,
  color: '#103B81',
  textAlign: 'left',
});

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginTop: 18,
    marginBottom: 18,
  },
  card: {
    width: '33%',
    minWidth: 251,
  },
  cardMiddle: {
    alignItems: 'center',
  },
  cardLast: {
    alignItems: 'flex-end',
  },
  imageText: {
    fontSize: 14,
    fontWeight: '300',
    fontFamily: 'Avenir',
    textAlign: 'center',
    width: 180,
    flex: 1,
    color: '#103B81',
  },
  description: {
    fontSize: 14,
    fontWeight: '300',
    color: '#103B81',
    marginLeft: 32,
  },
});

export default styles;
