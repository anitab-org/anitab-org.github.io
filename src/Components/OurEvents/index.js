import React from 'react';
import ImageContent from './../ImageContent';
import SectionHeader from './../SectionHeader';
import { getEvents } from './../../content/our_events';
import { Box, Content, Description } from './styles';


function OurEvents() {
  const content = getEvents();
  const renderContent = (index, section) => {
    return (
      <Content key={index}>
        <SectionHeader title={section.title} />
        {section.content.map((content, indx) => {
          return <Description key={indx}>{content.par}</Description>;
        })}
      </Content>
    );
  };

  return (
    <Box>
    {content.sections.map((section, index) => {
      return (
        <ImageContent key={index}
          image={section.image.source}
          imageSide={index % 2 === 0 ? 'right' : 'left'}
          Children={() => renderContent(index, section)}
        />
      );
    })}
    </Box>

  );
}
export default OurEvents;
