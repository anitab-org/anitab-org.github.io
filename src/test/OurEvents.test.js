import React from 'react';
import {render} from '@testing-library/react';
import OurEvents from './../Components/OurEvents';

it('should take a snapshot', () => {
    const { asFragment } = render(<OurEvents />);

    expect(asFragment(<OurEvents />)).toMatchSnapshot();
})