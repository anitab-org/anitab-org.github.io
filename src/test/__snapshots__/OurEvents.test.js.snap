// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-view-1dbjc4n"
    style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%; flex-direction: column; flex-wrap: wrap; margin-top: 32px; padding-left: 16px; padding-right: 16px; width: 80%;"
  >
    <div
      class="css-view-1dbjc4n r-alignItems-z6ln5t r-flex-13awgt0 r-flexDirection-1euycsn r-flexWrap-1w6e6rj r-marginTop-1v1z2uz"
    >
      <div
        class="css-view-1dbjc4n r-paddingLeft-1qhn6m8 r-paddingRight-i023vh"
      >
        <a>
          <div
            class="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
            style="width: 500px;"
          >
            <div
              class="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
              style="background-image: url(our_events.png);"
            />
            <img
              alt=""
              class="css-accessibilityImage-9pa8cd"
              draggable="false"
              src="our_events.png"
            />
          </div>
        </a>
      </div>
      <div
        class="css-view-1dbjc4n r-alignItems-z6ln5t r-flex-13awgt0 r-flexDirection-1fspl2j r-flexGrow-16y2uox r-minWidth-pfqljz"
      >
        <div
          class="css-view-1dbjc4n r-paddingLeft-1qhn6m8"
          style="flex-direction: column; margin-right: 77px;"
        >
          <div
            class="css-view-1dbjc4n r-flex-13awgt0 r-paddingLeft-1qhn6m8 r-paddingRight-i023vh r-paddingTop-9cjmro r-width-13qz1uu"
          >
            <div
              class="css-text-901oao r-alignItems-1habvwh r-color-1qovj5m r-flex-13awgt0 r-fontSize-1ui5ee8 r-fontWeight-16dba41"
              dir="auto"
            >
              EVENTS
            </div>
          </div>
          <div
            class="css-text-901oao r-color-1qovj5m r-flex-13awgt0 r-flexGrow-16y2uox r-marginTop-1v1z2uz r-paddingLeft-1qhn6m8"
            dir="auto"
            style="font-size: 18px; font-weight: 200; text-align: left;"
          >
            We at AnitaB.org Open Source believe in Inspiration through Action. To facilitate our Open Source journey and continue the AnitaB.org legacy we conduct enthralling events all year long, with the purpose of bridging the gap between creative developers and our Open Source community. Keep scrolling to catch a glimpse of few such events! 
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-view-1dbjc4n r-alignItems-z6ln5t r-flex-13awgt0 r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-marginTop-1v1z2uz"
    >
      <div
        class="css-view-1dbjc4n r-paddingLeft-1qhn6m8 r-paddingRight-i023vh"
      >
        <a>
          <div
            class="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
            style="width: 500px;"
          >
            <div
              class="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
              style="background-image: url(our_events.png);"
            />
            <img
              alt=""
              class="css-accessibilityImage-9pa8cd"
              draggable="false"
              src="our_events.png"
            />
          </div>
        </a>
      </div>
      <div
        class="css-view-1dbjc4n r-alignItems-z6ln5t r-flex-13awgt0 r-flexDirection-1fspl2j r-flexGrow-16y2uox r-minWidth-pfqljz"
      >
        <div
          class="css-view-1dbjc4n r-paddingLeft-1qhn6m8"
          style="flex-direction: column; margin-right: 77px;"
        >
          <div
            class="css-view-1dbjc4n r-flex-13awgt0 r-paddingLeft-1qhn6m8 r-paddingRight-i023vh r-paddingTop-9cjmro r-width-13qz1uu"
          >
            <div
              class="css-text-901oao r-alignItems-1habvwh r-color-1qovj5m r-flex-13awgt0 r-fontSize-1ui5ee8 r-fontWeight-16dba41"
              dir="auto"
            >
              TITLE 2
            </div>
          </div>
          <div
            class="css-text-901oao r-color-1qovj5m r-flex-13awgt0 r-flexGrow-16y2uox r-marginTop-1v1z2uz r-paddingLeft-1qhn6m8"
            dir="auto"
            style="font-size: 18px; font-weight: 200; text-align: left;"
          >
            We at AnitaB.org Open Source believe in Inspiration through Action. To facilitate our Open Source journey and continue the AnitaB.org legacy we conduct enthralling events all year long, with the purpose of bridging the gap between creative developers and our Open Source community. Keep scrolling to catch a glimpse of few such events! 
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
